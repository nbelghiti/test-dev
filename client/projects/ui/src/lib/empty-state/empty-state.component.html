<div class="c-empty-state" [ngClass]="{
	'c-empty-state--small': size === EmptyStateSize.small,
	'c-empty-state--large': size === EmptyStateSize.large
}">
	<span class="c-empty-state__icon" [ngClass]="{
		'c-empty-state__icon--small': smallIcon || size === EmptyStateSize.small
	}">
		<i class="fas fa-{{ icon }}"></i>
	</span>

	<p class="c-empty-state__message">{{ message }}</p>

	<div #subMessage [ngClass]="{
		'c-empty-state__sub-message': subMessage.childNodes.length !== 0
	}">
		<ng-content></ng-content>
	</div>
</div>
