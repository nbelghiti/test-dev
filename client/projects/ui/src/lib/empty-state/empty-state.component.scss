// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: Empty state
// =============================================================================
.c-empty-state {
	display: flex;
	flex-direction: column;
	justify-content: center;
	align-items: center;
	padding: 2rem;

	text-align: center;
}

// =============================================================================
// :: Elements
// =============================================================================
.c-empty-state__icon {
	font-size: 5.2rem;

	color: $text-color-grey-76;
}

.c-empty-state__message {
	margin-top: 2rem;

	font-size: $font-size-1x-large;
	font-weight: $font-weight-bold;

	color: $text-color-grey-46;
}

.c-empty-state__sub-message {
	margin-top: 1rem;

	font-size: $font-size-base;

	color: $text-color-grey-46;
}

// =============================================================================
// :: Modifiers
// =============================================================================
.c-empty-state__icon--small {
	font-size: 3.6rem;

	+ .c-empty-state__message {
		margin-top: 1rem;
	}
}

.c-empty-state__icon--large {
	font-size: 10rem;

	+ .c-empty-state__message {
		margin-top: 4rem;
	}
}

.c-empty-state--small {

	.c-empty-state__message {
		font-size: $font-size-base;
	}

	.c-empty-state__sub-message {
		font-size: $font-size-1x-small;
	}
}

.c-empty-state--large {

	.c-empty-state__message {
		font-size: $font-size-2x-large;
	}

	.c-empty-state__sub-message {
		font-size: $font-size-1x-large;
	}
}
