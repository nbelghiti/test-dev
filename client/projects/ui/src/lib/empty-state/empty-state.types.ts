export enum EmptyStateSize {
	small = 'small',
	normal = 'normal',
	large = 'large',
}
