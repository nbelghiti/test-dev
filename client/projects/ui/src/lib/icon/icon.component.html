<i class="c-icon__glyph {{ size && 'c-icon__glyph--' + size }} {{ type && 'c-icon__glyph--' + type }} fas fa-{{ name }}"></i>
