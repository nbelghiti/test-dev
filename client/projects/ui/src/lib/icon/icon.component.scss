// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: Icon
// =============================================================================
:host {
	display: inline-flex;
	justify-content: center;
	align-items: center;

	font-size: inherit;
}

// =============================================================================
// :: Elements
// =============================================================================
.c-icon__glyph {
	line-height: inherit;

	cursor: pointer;
}

// =============================================================================
// :: Modifiers
// =============================================================================
.c-icon__glyph--large {
	font-size: 2rem;
}

.c-icon__glyph--small {
	font-size: 1.4rem;
}

.c-icon__glyph--tiny {
	font-size: 1.2rem;
}

.c-icon__glyph--success {
	color: $ui-color-success-dark;
}

.c-icon__glyph--error {
	color: $ui-color-error-dark;
}

.c-icon__glyph--warning {
	color: $ui-color-warning-dark;
}

.c-icon__glyph--info {
	color: $ui-color-info-dark;
}
