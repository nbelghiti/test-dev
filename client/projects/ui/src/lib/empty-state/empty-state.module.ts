import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';

import { EmptyStateComponent } from './empty-state.component';

@NgModule({
	imports: [
		CommonModule,
	],
	declarations: [
		EmptyStateComponent,
	],
	exports: [
		EmptyStateComponent,
	],
})
export class EmptyStateModule {}
