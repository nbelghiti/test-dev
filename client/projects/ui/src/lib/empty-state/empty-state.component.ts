import { Component, Input } from '@angular/core';

import { IconName } from '../icon/icon.types';

import { EmptyStateSize } from './empty-state.types';

@Component({
	selector: 'ui-empty-state',
	templateUrl: './empty-state.component.html',
	styleUrls: ['./empty-state.component.scss']
})

export class EmptyStateComponent {
	@Input() public icon: IconName;
	@Input() public message: string;
	@Input() public smallIcon: boolean = false;
	@Input() public size: EmptyStateSize = EmptyStateSize.normal;

	public EmptyStateSize = EmptyStateSize;
}
