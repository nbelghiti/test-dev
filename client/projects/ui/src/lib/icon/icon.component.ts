import { Component, HostBinding, Input } from '@angular/core';

import { IconName, IconSize, IconType } from './icon.types';

@Component({
	selector: 'ui-icon',
	templateUrl: './icon.component.html',
	styleUrls: ['./icon.component.scss']
})

export class IconComponent {
	@Input() public name: IconName;
	@Input() public type: IconType = null;
	@Input() public size: IconSize = null;

	@HostBinding('class.c-icon') public className = true;
}
