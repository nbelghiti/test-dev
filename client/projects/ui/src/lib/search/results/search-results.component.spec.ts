import { ComponentFixture, TestBed, async } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';

import { EmptyStateComponent } from '../../empty-state/empty-state.component';
import { IconComponent } from '../../icon/icon.component';
import { SearchResultComponent } from '../result/search-result.component';

import { SearchResultsComponent } from './search-results.component';

describe('SearchResultsComponent', () => {
	let component: SearchResultsComponent;
	let fixture: ComponentFixture<SearchResultsComponent>;

	beforeEach(async(() => {
		TestBed.configureTestingModule({
			imports: [
				RouterTestingModule
			],
			declarations: [
				SearchResultsComponent,
				SearchResultComponent,
				EmptyStateComponent,
				IconComponent,
			]
		})
		.compileComponents();
	}));

	beforeEach(() => {
		fixture = TestBed.createComponent(SearchResultsComponent);
		component = fixture.componentInstance;
		fixture.detectChanges();
	});

	it('should create', () => {
		expect(component).toBeTruthy();
	});

	it('Should render correctly', () => {
		expect(fixture).toMatchSnapshot();
	});
});
