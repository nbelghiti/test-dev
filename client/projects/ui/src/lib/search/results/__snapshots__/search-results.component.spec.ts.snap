// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResultsComponent Should render correctly 1`] = `
<ui-search-results
  IconName={[Function Object]}
  SearchResultType={[Function Object]}
  active="false"
  items={[Function QueryList]}
  navigate={[Function Boolean]}
  selectResult={[Function EventEmitter]}
>
  <ui-empty-state
    ng-reflect-icon="search"
    ng-reflect-message="We didn't find any undefined f"
    ng-reflect-small-icon="true"
  >
    <div
      class="c-empty-state"
      ng-reflect-klass="c-empty-state"
      ng-reflect-ng-class="[object Object]"
    >
      <span
        class="c-empty-state__icon c-empty-state__icon--small"
        ng-reflect-klass="c-empty-state__icon"
        ng-reflect-ng-class="[object Object]"
      >
        <i
          class="fas fa-search"
        />
      </span>
      <p
        class="c-empty-state__message"
      >
        We didn't find any undefined for your search
      </p>
      <div
        class="c-empty-state__sub-message"
        ng-reflect-ng-class="[object Object]"
      >
         Try changing your searchterm (in: project name, component name, inputfield...) 
      </div>
    </div>
  </ui-empty-state>
</ui-search-results>
`;
